app/gui/view/src/debug_mode_popup.rs

```diff
@@ -0,0 +1,208 @@
+//! Text message on top of the screen that signals about enabling/disabling Debug Mode of Graph
+//! Editor.
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl::animation::delayed::DelayedAnimation;
+use ensogl::application::Application;
+use ensogl::display;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::*;
+use ensogl::Animation;
+use ensogl_component::label::Label;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// Mitigate limitations of constant strings concatenation.
+macro_rules! define_debug_mode_shortcut {
+    ($shortcut:literal) => {
+        /// A keyboard shortcut used to enable/disable Debug Mode.
+        pub const DEBUG_MODE_SHORTCUT: &str = $shortcut;
+        const DEBUG_MODE_ENABLED: &str =
+            concat!("Debug Mode enabled. To disable, press `", $shortcut, "`.");
+    };
+}
+define_debug_mode_shortcut!("ctrl shift d");
+const DEBUG_MODE_DISABLED: &str = "Debug Mode disabled.";
+
+const LABEL_VISIBILITY_DELAY_MS: f32 = 3_000.0;
+const LABEL_PADDING_TOP: f32 = 50.0;
+
+
+
+// ==================
+// === PopupLabel ===
+// ==================
+
+/// Text label that disappears after a predefined delay.
+#[derive(Debug, Clone, CloneRef)]
+struct PopupLabel {
+    label:   Label,
+    network: frp::Network,
+    show:    frp::Source,
+    hide:    frp::Source,
+}
+
+impl display::Object for PopupLabel {
+    fn display_object(&self) -> &display::object::Instance<Scene> {
+        self.label.display_object()
+    }
+}
+
+impl PopupLabel {
+    /// Constructor. Label will have `text` content and will disappear after `delay` milliseconds.
+    pub fn new(app: &Application, text: &str, delay: f32) -> Self {
```

Comment:
So far we've been minimizing the constructor arguments of visual components: they should have some default (e.g. an empty text) and have a separate `set_text` method. You can also make a separate `new_with_text_and_delay` method.

Reply:
Done